packages/block-library/src/term-description/index.php

```diff
@@ -13,8 +13,9 @@
  * @return string Returns the filtered post content of the current post.
  */
 function render_block_core_term_description( $attributes ) {
+	$term_description = term_description();
```

Comment:
I think it would be more efficient to move this to after the first set of tests. `term_description()` performs these tests already when the term_id is not passed in.

Reply:
I did not understand.

Reply:
Both this function and `term_description()` perform `is_category()`, `is_tag()` and `is_tax()` tests. If you run term_description first then there's no need to run the tests again, you just need to test for empty( $term_description ) and return early. 

If you rely on `term_description()` to perform these tests it introduces an unnecessary overhead if the block is not used in the correct context.  The code will attempt to access information about a term that shouldn't exist. You'll get the same end result on the front end but will have triggered WP_Error logic unnecessarily.

This is v. minor in the great scale of things.

Reply:
I am sorry I still do not understand. You are first asking me to "move this **after** the first set of tests" then you are telling me that if I run it **first** I don't need to run it again but then you follow that up by saying I should not rely on it.

Perhaps it will be easier for me if you add the code example.

Reply:
Are you asking me to do this?

```
	if ( ! is_category() && ! is_tag() && ! is_tax() ) {
		return '';
	}

	$term_description = term_description();

	if ( empty( $term_description )  ) {
		return '';
	}
```

Reply:
If I understand correctly, this is what @bobbingwide is saying: https://github.com/WordPress/gutenberg/pull/30564#pullrequestreview-631091278

`term_description()` already runs the checks, and if not a category or not a tag or not a taxonomy term, `$term_description` will be empty, so we can simplify that check

Reply:
is that not the scenario where "it introduces an unnecessary overhead if the block is not used in the correct context ... ...will have triggered WP_Error logic unnecessarily."

Reply:
True... 🤔

Reply:
I like https://github.com/WordPress/gutenberg/pull/30564#discussion_r609423425 but would personally prefer 
```
$term_description = '';
if ( is_category() || is_tag() || is_tax() ) {
   $term_description = term_description();
}
if ( empty( $term_description) ) {
   return '';
}
```
This should prevent `term_description()` being asked to do dumb things.